tests/bdd/features/volume_group/create/test_create.py

```diff
@@ -0,0 +1,258 @@
+    parsers,
+)
+
+import openapi.exceptions
+from common.apiclient import ApiClient
+from common.deployer import Deployer
+from openapi.model.create_pool_body import CreatePoolBody
+from openapi.model.create_volume_body import CreateVolumeBody
+from openapi.model.volume_policy import VolumePolicy
+
+NODE_NAME_1 = "io-engine-1"
+NODE_NAME_2 = "io-engine-2"
+NODE_NAME_3 = "io-engine-3"
+NON_VG_VOLUME_UUID = "5cd5378e-3f05-47f1-a830-a0f5873a1449"
+VG_VOLUME_UUID_1 = "c36f7453-ae65-43bb-8ee9-4b5a373a8ed4"
+VG_VOLUME_UUID_2 = "07805245-ec1c-43bd-a950-c42f4c3a9ac7"
+VG_VOLUME_UUID_3 = "6805e636-1986-4ab7-95d0-75597df1532b"
+VOLUME_SIZE = 10485761
+
+
+@scenario("create.feature", "creating volume group with 3 volumes with 1 replica each")
+def test_creating_volume_group_with_3_volumes_with_1_replica_each():
+    """creating volume group with 3 volumes with 1 replica each."""
+
+
+@scenario(
+    "create.feature",
+    "creating volume group with 3 volumes with 1 replica each with limited pools",
+)
+def test_creating_volume_group_with_3_volumes_with_1_replica_each_with_limited_pools():
+    """creating volume group with 3 volumes with 1 replica each with limited pools."""
+
+
+@scenario(
+    "create.feature", "creating volume group with 3 volumes with 3 replica on 3 pools"
+)
+def test_creating_volume_group_with_3_volumes_with_3_replica_on_3_pools():
+    """creating volume group with 3 volumes with 3 replica on 3 pools."""
+
+
+@scenario(
+    "create.feature", "creating volume group with 3 volumes with 3 replica on 5 pools"
+)
+def test_creating_volume_group_with_3_volumes_with_3_replica_on_5_pools():
+    """creating volume group with 3 volumes with 3 replica on 5 pools."""
+
+
+@scenario(
+    "create.feature", "creating volume group with 3 volumes with 3 replica on 9 pools"
+)
+def test_creating_volume_group_with_3_volumes_with_3_replica_on_9_pools():
+    """creating volume group with 3 volumes with 3 replica on 9 pools."""
+
+
+@given("2 pools, 1 on node A, 0 on node B, 1 on node C")
+def _():
+    """2 pools, 1 on node A, 0 on node B, 1 on node C."""
+    node_pool_map = {NODE_NAME_1: 1, NODE_NAME_2: 0, NODE_NAME_3: 1}
+    create_node_pools(node_pool_map)
+
+
+@given("3 pools, 1 on node A, 1 on node B, 1 on node C")
+def _():
+    """3 pools, 1 on node A, 1 on node B, 1 on node C."""
+    node_pool_map = {NODE_NAME_1: 1, NODE_NAME_2: 1, NODE_NAME_3: 1}
+    create_node_pools(node_pool_map)
+
+
+@given("5 pools, 2 on node A, 1 on node B, 2 on node C")
+def _():
+    """5 pools, 2 on node A, 1 on node B, 2 on node C."""
+    node_pool_map = {NODE_NAME_1: 2, NODE_NAME_2: 1, NODE_NAME_3: 2}
+    create_node_pools(node_pool_map)
+
+
+@given("9 pools, 3 on node A, 3 on node B, 3 on node C")
+def _():
+    """9 pools, 3 on node A, 3 on node B, 3 on node C."""
+    node_pool_map = {NODE_NAME_1: 3, NODE_NAME_2: 3, NODE_NAME_3: 3}
+    create_node_pools(node_pool_map)
+
+
+@given("an io-engine cluster")
+def _():
+    """an io-engine cluster."""
```

Comment:
does this mean we re-create the cluster every time? If so, can we use an autouse fixture here?

Reply:
Correct me if I am wrong, `autouse` will ensure this gets called everytime right for each scenario? Added.